### PR TITLE
[Merged by Bors] - Logging

### LINKED ISSUE #291: Enable log aggregation
Integrate log aggregation according to [ADR025: Logging and Log Aggregation Architecture](https://docs.stackable.tech/home/nightly/contributor/adr/ADR025-logging_architecture.html).

### PR DESCRIPTION
**Pull request by siegfriedweber**
# Description

Add log aggregation

Closes #291 

## Integration Tests

[Build Status](https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hbase-operator-it-custom/65/)

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/product_logging.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    #[snafu(display("failed to retrieve the ConfigMap {cm_name}"))]
+    ConfigMapNotFound {
+        source: stackable_operator::error::Error,
+        cm_name: String,
+    },
+    #[snafu(display("failed to retrieve the entry {entry} for ConfigMap {cm_name}"))]
+    MissingConfigMapEntry {
+        entry: &'static str,
+        cm_name: String,
+    },
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply rust/operator-binary/src/product_logging.rs]: changed

File: rust/operator-binary/src/hbase_controller.rs [hunk 2]
```diff
@@ -151,15 +171,24 @@ pub enum Error {
...
     #[snafu(display("failed to convert java heap config to unit [{unit}]"))]
     FailedToConvertJavaHeap {
         source: stackable_operator::error::Error,
         unit: String,
     },
+    #[snafu(display("failed to resolve the Vector aggregator address"))]
+    ResolveVectorAggregatorAddress {
+        source: crate::product_logging::Error,
+    },
+    #[snafu(display("failed to add the logging configuration to the ConfigMap {cm_name}"))]
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply rust/operator-binary/src/hbase_controller.rs]: changed

File: tests/templates/kuttl/logging/06-assert.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+metadata:
+  name: test-logs
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply tests/templates/kuttl/logging/06-assert.yaml]: Metadata removed

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by siegfriedweber** [comment]
bors merge

**Pull request closed without merging**